src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,15 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin().length() < 6
+                || user.getPassword().length() < 6
+                || user.getAge() < 18) {
+            throw new InvalidUserDataException("Invalid user data");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException("User with this login already exists");
+        }
```

Comment:
1. check fields for nulls before working with them
2. extract limits to constants
3. write more informative exception messages to let user know what exactly is wrong